third_party/move/move-vm/types/src/code/cache.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::hash::Hash;
+
+/// Interface used by any script cache implementation.
+pub trait ScriptCache {
+    type Key: Eq + Hash + Clone;
+    type Script: Clone;
+
+    /// Stores the script to the code cache.
+    fn store_script(&self, key: Self::Key, script: Self::Script);
```

Comment:
Let's clearly define (document) what happens if the script is already there (Should we return a bool, or a previous version?).

Also naming:
probably want `fn insert_script`, and `get_script`, as in insert to the cache, and get_from_cache.

Seme for below

Reply:
Good point, renamed. For return type, I think bool is enough here:
- For scripts, we probably do not want to have multiple arcs, and only a single one
- For modules the same, but also we can check module version inside, decide if we want to override or not, and return the boolean. The caller can refetch in this case.

Reply:
Actually, checking the code, we just need a ref to the item stored inside so that the caller can resolve. This way we also avoid arcs floating around.